### PR TITLE
Refine children parent communication

### LINKED ISSUE #270: Add child callback to handle messages from parent, replace forward action
Currently, the communication between parent and child looks like this:
* child -> action `notify` -> parent -> handle_notification
* parent -> action `forward` -> child -> handle_other
We need to:
* unify the naming here
* provide a callback in child other than `handle_other` to distinguish messages from parent and from other processes
* `forward` is quite confusing when used to deliver messages that originate in parent, we need a better name for this action

## Proposed solutions

* `handle_{child,parent}_message` + `send_{child,parent}` actions
* `handle_{child,parent}_notification + `notify_{child,parent}` actions`
*  Leave `notify` and `handle_notification`, add `handle_child_notification` and `notify_child` actions

Besides the above, forward may be removed or changed into mechanism allowing routing messages without the need to implement callbacks

@mat-hek What's your take on the names? Any other suggestions?

### PR DESCRIPTION
**Pull request by varsill**
closes #270

### CODE REVIEW

File: lib/membrane/core/bin.ex [hunk 1]
```diff
@@ -130,6 +140,22 @@ defmodule Membrane.Core.Bin do
...
+    state =
+      CallbackHandler.exec_and_handle_callback(
+        :handle_parent_notification,
+        Membrane.Core.Bin.ActionHandler,
+        %{context: context},
+        notification,
+        state
+      )
+
+    {:noreply, state}
```
Comment (mat-hek) [inline lib/membrane/core/bin.ex:156]: let's move this to `Membrane.Core.Child.LifecycleController`

**Review by mat-hek — APPROVED** [review]

File: lib/membrane/core/child/lifecycle_controller.ex [hunk 2]
```diff
@@ -0,0 +1,29 @@
+defmodule Membrane.Core.Child.LifecycleController do
+  @moduledoc false
+  alias Membrane.Core.{CallbackHandler, Component}
+
+  require Membrane.Core.Component
+
+  @type state_t ::
+          Membrane.Core.Bin.State.t()
+          | Membrane.Core.Element.State.t()
```
Comment (mat-hek) [inline lib/membrane/core/child/lifecycle_controller.ex:9]: Use `Membrane.Core.Child.state_t`

File: lib/membrane/core/child/lifecycle_controller.ex [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  @spec handle_parent_notification(Membrane.ParentNotification.t(), state_t) ::
+          state_t
+  def handle_parent_notification(notification, state) do
+    context = Component.callback_context_generator(:child, ParentNotification, state)
+
+    action_handler =
+      if Component.is_bin?(state),
```
Comment (mat-hek) [inline lib/membrane/core/child/lifecycle_controller.ex:17]: Use `Component.action_handler`

File: lib/membrane/core/child/lifecycle_controller.ex (hunk 2 above)
Comment (varsill) [reply lib/membrane/core/child/lifecycle_controller.ex:9]: done

File: lib/membrane/core/child/lifecycle_controller.ex (hunk 3 above)
Comment (varsill) [reply lib/membrane/core/child/lifecycle_controller.ex:17]: done

File: lib/membrane/core/element.ex [hunk 4]
```diff
@@ -20,8 +20,9 @@ defmodule Membrane.Core.Element do
 
   alias Membrane.{Clock, Element, Sync}
   alias Membrane.ComponentPath
-  alias Membrane.Core.Element.{LifecycleController, PadController, PlaybackBuffer, State}
-  alias Membrane.Core.{Message, PlaybackHandler, Telemetry, TimerController}
+  alias Membrane.Core.Element.LifecycleController, as: ElementLifecycleController
```
[the remark this answers is not in the document]
Comment (varsill) [reply lib/membrane/core/element.ex]: done

**Review by mat-hek — APPROVED** [review]

**Pull request merged**